clients/cobol-lsp-vscode-extension/src/services/ProcessorGroups.ts

```diff
@@ -286,3 +310,53 @@ async function loadProcessorGroupSettings<T extends string | string[]>(
     return configObject;
   }
 }
+
+export function prepareProcessorGroupConfigPathsForDsnAndUss(
+  pgConfigs: (ZoweDatasetConfigModel | ZoweUssConfigModel)[],
+): { path: string; profile?: string }[] {
+  const paths: { path: string; profile?: string }[] = [];
+  for (const config of pgConfigs) {
+    if (DATASET in config)
+      paths.push({
+        path: config.dataset,
+        profile: config.profile ? config.profile : undefined,
+      });
+    else if (USSFILE in config)
+      paths.push({
+        path: config.ussFile,
+        profile: config.profile ? config.profile : undefined,
+      });
+  }
+  return paths;
+}
+
+export async function prepareProcessorGroupConfigPathsForEndevor(
+  pgConfigs: EndevorConfigModel[],
+  e4eDownloader: CopybookDownloaderForE4E,
+  copybook: CopybookName,
+): Promise<
+  { element: EndevorElement; profile: ResolvedProfile }[] | undefined
+> {
+  const paths: { element: EndevorElement; profile: ResolvedProfile }[] = [];
+
+  for (const config of pgConfigs) {
+    let profile: Partial<ResolvedProfile>;
+    if (!config.profile) {
+      profile = { profile: undefined, instance: undefined };
+    } else {
+      profile = {
+        profile: config.profile.split("@")[1],
+        instance: config.profile.split("@")[0],
```

Comment:
The other way around ("endevorinstance@profile").
Also you want to split only on the first `@`.
Only instance `instance`.
Only profile `@profile`.